crates/proof-of-sql/src/sql/proof_plans/table_exec.rs

```diff
@@ -33,17 +33,22 @@ impl TableExec {
     }
 
     /// Returns the entire table.
-    fn get_table<'a, S: Scalar>(&self, accessor: &'a dyn DataAccessor<S>) -> Vec<Column<'a, S>> {
-        self.schema
-            .iter()
-            .map(|field| {
+    ///
+    /// # Panics
+    /// Panics if columns from the accessor have different lengths.
+    /// In practice, this should never happen.
+    fn get_table<'a, S: Scalar>(&self, accessor: &'a dyn DataAccessor<S>) -> Table<'a, S> {
+        Table::<'a, S>::try_from_iter(self.schema.iter().map(|field| {
+            (
+                field.name(),
                 accessor.get_column(ColumnRef::new(
                     self.table_ref,
                     field.name(),
                     field.data_type(),
-                ))
-            })
-            .collect()
+                )),
+            )
+        }))
+        .expect("Failed to create table from column references")
```

Comment:
This is problematic if the schema is empty.
Why not just use `accessor.get_table`, which already handles this?